interlok-core/src/main/java/com/adaptris/core/MultiPayloadAdaptrisMessageImp.java

```diff
@@ -651,13 +650,10 @@ private class Payload {
     }
 
     Payload(@NotNull ByteArrayOutputStream stream) {
-      this.stream = stream;
+      this.data = stream.toByteArray();
```

Comment:
I think, if you go this way, you may as well change line 629 ` payloads.put(payloadId, new Payload(out));`
by something like line 635 `addPayload(payloadId, ((ByteArrayOutputStream) super.out).toByteArray());`

Then you can remove this constructor altogether as it was only used line 629.

I'm not really sure why it was doing that but it may worth doing more tests with `MultiPayloadStreamOutputParameter` and `MultiPayloadXmlTransformService` as they are using `msg.getOutputStream(outputPayloadId)` which eventually call the code we're talking about.

In fairness, I don't really know why this code was like that and if we are not breaking something removing it. But I'd otherwise agree with your point.

Reply:
I reviewed the classes you mentioned and it looks like doni what you suggested doesn't break these.

I had a look at what the test classes for these are doing and in the tests I can see they are taking an input stream, writing it to the messages output stream and then adding it as a new payload.

After making the changes these tests are still working as expected so is it safe to say that it is worknig as expected as it still successfully writes to the messages output stream AND is able to add it as a new payload to the multi payload message, it can also retrive the payload to check it's content.